docker/build-image.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+#  Copyright (C) 2018-2021 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+USERNAME=usdotfhwastol
+
+cd "$(dirname "$0")"
+IMAGE=$(./get-image-name.sh)
+
+echo ""
+echo "##### $IMAGE Docker Image Build Script #####"
+echo ""
+
+while [[ $# -gt 0 ]]; do
+    arg="$1"
+    case $arg in
+        -v|--version)
+            COMPONENT_VERSION_STRING="$2"
+            shift
+            shift
+            ;;
+        --system-release)
+            SYSTEM_RELEASE=true
+            shift
+            ;;
+        -p|--push)
+            PUSH=true
+            shift
+            ;;
+        -d|--develop)
+            USERNAME=usdotfhwastoldev
+            COMPONENT_VERSION_STRING=develop
+            shift
+            ;;
+    esac
+done
+
+if [[ -z "$COMPONENT_VERSION_STRING" ]]; then
+    COMPONENT_VERSION_STRING=$("./get-component-version.sh")
+fi
+
+echo "Building docker image for $IMAGE version: $COMPONENT_VERSION_STRING"
+echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING"
+
+cd ..
+if [[ $COMPONENT_VERSION_STRING = "develop" ]]; then
+    sed "s|usdotfhwastoldev/|$USERNAME/|g; s|usdotfhwastolcandidate/|$USERNAME/|g; s|usdotfhwastol/|$USERNAME/|g; s|:[0-9]*\.[0-9]*\.[0-9]*|:$COMPONENT_VERSION_STRING|g; s|checkout.bash|checkout.bash -d|g" \
```

Comment:
Remove this sed operation. This is in reference to a checkout.bash script you do not yet have. Also sed operations on version control files is bad practice because it is error prone and very hard to diagnose issues with especially in CI/CD operations. Instead we have moved to using branch arguments for the checkout.bash script (https://github.com/usdot-fhwa-stol/carma-msgs/blob/develop/docker/checkout.bash) which can be passed to Dockerfile as argument like (https://github.com/usdot-fhwa-stol/carma-msgs/blob/2c6bb075f5301ba490b60adacafaa749401036ec/Dockerfile#L10)